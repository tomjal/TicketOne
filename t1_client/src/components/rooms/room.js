import React, { Component } from 'react';

import ChatWindow from './chatWindow';

class Room extends Component {
  constructor() {
    super();
    this.state = { isOpened: true };
  }
  sendMessage(message) {
    this.props.sendMessageCallback(message, this.props.id);
  }
  isClientRoom(role) {
    if (role == "client") {
      return true;
    }
    return false;
  }
  render() {
    const { id, role, messages } = this.props;
    const isClientRoom = this.isClientRoom(role);
    return (
      <div className="room">
        {isClientRoom && <div className="widget-header-text">Hello {id} - how can we help you?</div>}
        {!isClientRoom && <div className="widget-header-text">Room with client - {id}</div>}
        <div>
          {this.state.isOpened &&
            <ChatWindow
              role={role}
              sendMessage={this.sendMessage.bind(this)}
              messages={messages} />}
        </div>
      </div>
    );
  }
}

export default Room;
