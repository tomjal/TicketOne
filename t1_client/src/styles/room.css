/*
* === Chatroom specific style ===
*/

.room-flex-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}

.room {
  width: 450px;
  height: 360px;
  padding-top: 10px;
  padding-bottom: 10px;
  background: white;
  border-radius: 4px;
  overflow: hidden;
  box-shadow: 0 1px 1px rgba(0,0,0,0.16), 0 2px 4px rgba(0,0,0,0.23);
}

.room-flex-filling {
  width: 480px;
  height: 0;
}

.room-chat-body {
  box-shadow: rgba(0, 0, 0, 0.2) 0px 0px 1px 0px, rgba(0, 0, 0, 0.1) 0px 1px 5px 0px;
}

.room-chat-list {
  width: 450px;
  height: 400px;
  overflow: hidden;
}

.room-chat-input {
  width: 450px;
  height: 400px;
  overflow: hidden;
}
