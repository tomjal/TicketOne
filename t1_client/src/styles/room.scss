@import './_colors.scss';
$room-width: 450px;
$room-height: 360px;
$room-flex-margin: 15px;
$room-box-shadow: 0 1px 1px rgba(0, 0, 0, 0.16),
0 2px 4px rgba(0, 0, 0, 0.23);
$room-chat-body-shadow: rgba(0, 0, 0, 0.2) 0px 0px 1px 0px,
rgba(0, 0, 0, 0.1) 0px 1px 5px 0px;

/*
* === Chatroom specific style ===
*/

.room-widget {
    width: $room-width;
    height: $room-height;
    padding-top: 10px;
    padding-bottom: 10px;
    overflow: hidden;
    background: $primary-text-color;
    border-radius: 4px;
    box-shadow: $room-box-shadow;
}

.room-flex-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
}

.room-flex-container>div {
    margin: $room-flex-margin;
}

.room-flex-container__filling {
    width: $room-width;
    height: 0;
    margin-top: 0!important;
    margin-bottom: 0!important;
}

.room-chat-body {
    box-shadow: $room-chat-body-shadow;
}

.room-chat-list {
    width: $room-width;
    height: calc(#{$room-height} + #{40px});
    overflow: hidden;
}