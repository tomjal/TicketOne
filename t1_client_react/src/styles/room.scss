@import './_colors.scss';
$room-width: 450px;
$room-margin: 15px;
$room-height: 400px;
$room-flex-margin: 15px;
$room-box-shadow: 0 1px 1px rgba(0, 0, 0, 0.16),
0 2px 4px rgba(0, 0, 0, 0.23);

/*
* === Room specific style ===
*/

.room-widget {
    width: $room-width;
    height: $room-height;
    margin: $room-margin;
    padding-top: 10px;
    padding-bottom: 10px;
    overflow: hidden;
    background: $primary-text-color;
    border-radius: 4px;
    box-shadow: $room-box-shadow;
}

.room-flex-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
}

.room-flex-container__filler {
    width: $room-width + 2 * $room-margin;
    height: 0;
    margin-top: 0!important;
    margin-bottom: 0!important;
}
