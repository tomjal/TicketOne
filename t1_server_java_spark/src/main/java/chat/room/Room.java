package app.room;

import lombok.*;

@Value
public class Book {
    
}