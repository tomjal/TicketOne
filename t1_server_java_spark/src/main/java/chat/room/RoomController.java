package app.room;

import app.login.*;
import app.util.*;
import spark.*;
import java.util.*;
import static app.util.JsonUtil.*;
import static app.util.RequestUtil.*;

public class RoomController {

    public static Route fetchAllRooms = (Request request, Response response) -> {
        
        if (clientAcceptsHtml(request)) {
            HashMap<String, Object> model = new HashMap<>();
            model.put("rooms", "");
        }
    };
}