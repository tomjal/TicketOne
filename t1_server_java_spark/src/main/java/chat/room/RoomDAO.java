package app.room;

import com.google.common.collect.*;
import java.util.*;

public class RoomDao {

    private final List<Room> rooms = ImmutableList.of(
    );

    public Iterable<Room> getAllRooms() {
        return rooms;
    }

    public Room getBookById(String id) {
        return books.stream().filter(b -> b.getId().equals(id)).findFirst().orElse(null);
    }
}