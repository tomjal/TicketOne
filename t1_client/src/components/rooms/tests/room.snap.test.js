import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Room } from './../room';

test('Room snapshot', () => {
  const wrap = shallow(<Room />);

  expect(shallowToJson(wrap)).toMatchSnapshot();
});