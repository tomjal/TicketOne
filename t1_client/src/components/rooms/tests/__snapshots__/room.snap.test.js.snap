exports[`test Room snapshot 1`] = `
<div
  className="room">
  <div
    className="widget-header-text">
    id: 
  </div>
  <div>
    <ChatWindow
      sendMessage={[Function]} />
  </div>
</div>
`;
